src/components/editor-page/editor-pane/linter/single-line-regex-linter.spec.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { SingleLineRegexLinter } from './single-line-regex-linter'
+import type { Diagnostic } from '@codemirror/lint'
+import { mockI18n } from '../../../markdown-renderer/test-utils/mock-i18n'
+import type { EditorView } from '@codemirror/view'
+import { Mock } from 'ts-mockery'
+import type { EditorState, Text } from '@codemirror/state'
+
+export const mockEditorView = (content: string): EditorView => {
+  const docMock = Mock.of<Text>()
+  docMock.toString = () => content
+  return Mock.of<EditorView>({
+    state: Mock.of<EditorState>({
+      doc: docMock
+    }),
+    dispatch: jest.fn()
+  })
+}
+
+const testSingleLineRegexLinter = (
+  regex: RegExp,
+  replace: (match: string) => string,
+  content: string,
+  diagnostics: Partial<Diagnostic>[]
+): void => {
+  const singleLineRegexLinter = new SingleLineRegexLinter(regex, 'message', replace, 'actionLabel')
+  const editorView = mockEditorView(content)
+  const calculatedDiagnostics = singleLineRegexLinter.lint(editorView)
+  expect(calculatedDiagnostics).toHaveLength(diagnostics.length)
+  for (let x = 0; x < calculatedDiagnostics.length; x++) {
```

Comment:
*screams in iteration via index*

How about a `.forEach()`? Like code from 2022? :sweat_smile: